src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -124,18 +113,98 @@ string IDkmClrResultProvider.GetUnderlyingString(DkmEvaluationResult result)
+                        inspectionContext,
+                        dataItem.Value,
+                        dataItem.Name,
+                        typeName: string.Empty,
+                        display: null,
+                        dataItem: dataItem));
+                    break;
+                case ExpansionKind.RawView:
+                    completionRoutine(CreateEvaluationResult(
+                        inspectionContext,
+                        dataItem.Value,
+                        Resources.RawView,
+                        typeName: string.Empty,
+                        display: null,
+                        dataItem: dataItem));
+                    break;
+                case ExpansionKind.ResultsView:
+                    completionRoutine(CreateEvaluationResult(
+                        inspectionContext,
+                        dataItem.Value,
+                        dataItem.Name,
+                        typeName: string.Empty,
+                        display: Resources.ResultsViewValueWarning,
+                        dataItem: dataItem));
+                    break;
+                case ExpansionKind.TypeVariables:
+                    var value = dataItem.Value;
+                    completionRoutine(DkmSuccessEvaluationResult.Create(
+                        inspectionContext,
+                        stackFrame,
+                        dataItem.Name,
+                        dataItem.FullName,
+                        dataItem.Flags,
+                        dataItem.DisplayValue,
+                        EditableValue: null,
+                        Type: dataItem.DisplayValue,
+                        Category: dataItem.Category,
+                        Access: value.Access,
+                        StorageType: value.StorageType,
+                        TypeModifierFlags: value.TypeModifierFlags,
+                        Address: value.Address,
+                        CustomUIVisualizers: null,
+                        ExternalModules: null,
+                        DataItem: dataItem));
+                    break;
+                default:
+                    Debug.Assert((dataItem.Kind == ExpansionKind.Default) || (dataItem.Kind == ExpansionKind.PointerDereference));
+                    // This call will evaluate DebuggerDisplayAttributes.
+                    GetResultAndContinue(
+                        dataItem,
+                        workList,
+                        declaredType: DkmClrType.Create(dataItem.Value.Type.AppDomain, dataItem.DeclaredType),
+                        inspectionContext: inspectionContext,
+                        parent: dataItem.Parent,
+                        completionRoutine: result => ContinueWithExceptionHandling(
+                            () => completionRoutine(result),
+                            e => completionRoutine(CreateEvaluationResultFromException(e, inspectionContext, stackFrame))));
+                    break;
```

Comment:
No need for ContinueWithExceptionHandling, this can be completionRoutine: completionRoutine.